src/system/user/gnwkbdm/manager.c

```diff
@@ -0,0 +1,86 @@
+//
+//  manager.c
+//  GunwOS
+//
+//  Created by Artur Danielewski on 08.12.2023.
+//
+
+#include <gunwipc.h>
+#include <gunwfug.h>
+#include <gunwrlp.h>
+#include <kbdmgr.h>
+
+#include "keyboard.h"
+#include "session.h"
+
+static void ipcAttachToKeyboardListener(const struct gnwIpcEndpointQuery * const query) {
+    if (!query) { 
+        fug(FUG_NULLPTR); 
+        return; 
+    }
+    if (!query->dataPtr) {
+        fug(FUG_INCONSISTENT);
+        return;
+    }
+    if (query->dataSizeBytes != sizeof(struct gnwKeyboardManagerAttachToKeyboardQuery)) {
+        fug(FUG_INCONSISTENT);
+        return;
+    }
+    if (query->replySizeBytes != sizeof(struct gnwKeyboardManagerAttachToKeyboardResult)) {
+        fug(FUG_INCONSISTENT);
+        return;
+    }
+
+    struct gnwKeyboardManagerAttachToKeyboardQuery * const kbdQueryPtr = (struct gnwKeyboardManagerAttachToKeyboardQuery *)query->dataPtr;
+    (void)kbdQueryPtr;
```

Comment:
?

Reply:
Mam co do tego fragmentu pewne wątpliwości - z jednej strony chciałbym z góry założyć, że to API może się odrobinę rozszerzyć, wówczas nie ryzykowałbym, że zapomnę o fakcie istnienia tej struktury. Z drugiej strony wygląda to dziwnie i nic nie wnosi.

Usunę - in progress

Reply:
Gotowe